### ISSUE TITLE
Autocorrect failing for unneeded condition cop with the following example code. 

### ISSUE THREAD

**Issue by jhirn** [rubocop-hq/rubocop#6080]
Hey all. I was running into the following error when running `rubocop -a`. Now this code is terrible and deserves to be shot (I wrote it), but tracking down which file was causing the error was a bit of a binary search. 

The code:

```
def predicate_method?
    object.some_boolean.present? if object.some_boolean.present? 
end
```

The exception: 
```
undefined method `type' for nil:NilClass
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cop/style/unneeded_condition.rb:76:in `else_source'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cop/style/unneeded_condition.rb:49:in `block in autocorrect'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cop/corrector.rb:61:in `block (2 levels) in rewrite'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/parser-2.5.1.0/lib/parser/source/tree_rewriter.rb:220:in `transaction'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cop/corrector.rb:60:in `block in rewrite'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cop/corrector.rb:58:in `each'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cop/corrector.rb:58:in `rewrite'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cop/team.rb:126:in `autocorrect_all_cops'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cop/team.rb:70:in `autocorrect'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cop/team.rb:98:in `block in offenses'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cop/team.rb:115:in `investigate'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cop/team.rb:94:in `offenses'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cop/team.rb:44:in `inspect_file'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:265:in `inspect_file'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:212:in `block in do_inspection_loop'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:244:in `block in iterate_until_no_changes'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:237:in `loop'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:237:in `iterate_until_no_changes'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:208:in `do_inspection_loop'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:111:in `block in file_offenses'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:129:in `file_offense_cache'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:109:in `file_offenses'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:100:in `process_file'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:78:in `block in each_inspected_file'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:75:in `each'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:75:in `reduce'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:75:in `each_inspected_file'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:67:in `inspect_files'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/runner.rb:39:in `run'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cli.rb:160:in `execute_runner'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cli.rb:88:in `execute_runners'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/lib/rubocop/cli.rb:41:in `run'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/exe/rubocop:13:in `block in <top (required)>'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/2.5.0/benchmark.rb:308:in `realtime'
/Users/jhirn/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rubocop-0.57.2/exe/rubocop:12:in `<top (required)>'
/Users/jhirn/.rbenv/versions/2.5.0/bin/rubocop:23:in `load'
/Users/jhirn/.rbenv/versions/2.5.0/bin/rubocop:23:in `<main>'
```
As requested the rubocop version:
``` 
!  2.5.0  ~/s/k/dot-com   master *~+  rubocop -V                                                                                    Fri Jul  6 11:51:41 2018
0.57.2 (using Parser 2.5.1.0, running on ruby 2.5.0 x86_64-darwin17)```

**Comment by koic** [comment]
This issue has been fixed in #5988 and will not be reproduced in master code of RuboCop. Thank you!

**Comment by jhirn** [comment]
No, thank you!!!! 🙏
